pkg/policy/api/ingress.go

```diff
@@ -99,6 +101,31 @@ type IngressCommonRule struct {
 	aggregatedSelectors EndpointSelectorSlice `json:"-"`
 }
 
+// DeepEqual returns true if both IngressCommonRule are deep equal.
+// The semantic of a nil slice in one of its fields is different from the semantic
+// of an empty non-nil slice, thus it explicitly checks for that case before calling
+// the autogenerated method.
+func (in *IngressCommonRule) DeepEqual(other *IngressCommonRule) bool {
+	if in.FromEndpoints == nil && other.FromEndpoints != nil ||
```

Comment:
This seems very repetitive. Perhaps a convenience  aka "isNilSlice" could help?

Reply:
I've added a `XorNil` helper in `pkg/slices`, see https://github.com/cilium/cilium/pull/33506/commits/f2301b56451d96143644ec680be55fd0a9962de8

Reply:
Brilliant! Thank you Fabio!